### ISSUE TITLE
a tool to edit stockpile presets, with operations like "subtract", "add" and "invert selection"

### ISSUE THREAD

**Issue by TaxiService** [DFHack/dfhack#3376]
what do i mean by this?

Imagine we had presets for, say: plantable seeds, cookable seeds, edible seeds, above-ground seeds, underground seeds... etc

now, this tool would let me do this:
- make a new seeds stockpile
- subtract the "plantable seeds" preset
- invert the selection, so now the pile is "all un-plantable seeds"
- subtract the "cookable seeds" preset
- subtract the "edible seeds" preset

now i've got a stockpile with only seeds that i can't either plant, cook or eat. °-° and i can set it to autotrade or something...

**Comment by TaxiService** [comment]
another thing that would be awesome would be this:
while we're editing a stockpile, or a stockpile preset, we can enter a mode that would let us click on another stockpile in the world, and do operations with _it_.

like, as an example.
- i make a stockpile for iron bars, one for copper bars, one for zinc.
- i make a fourth stockpiles for all metal bars.
- i can then select the all-metals 'pile, enter "subtract mode", and click on the other three.

this gives me a stockpile for "all other metals"

**Comment by Dragoon209** [comment]
Correct, the biome specific seeds that would be plant-able in the fort you have loaded.  Looking through the stockpiles command presets, it looks like there are some that may work already (shame on me for not looking first).  I'll have to check those out and see if they work like I want.  

It would be nice however to account for modded plants like you mentioned, and just have a stockpile filter tag or something like 'seeds:usable' or 'seeds:biome'.  I don't know enough about what is possible, so apologies if these are ridiculous requests.

**Comment by TaxiService** [comment]
oh dont worry! I don't know about what's possible either! :rofl:
or rather, i guess technically _everything_ is possible. 🤔 maybe sometimes it is not _feasible_, but asking for functionality like you do is a great way to find out!

**Comment by myk002** [comment]
Dynamic filtering is something I have planned for the `stockpiles` plugin as I slowly mold it into shape to support `quickfort`-driven stockpile configuration. I'll do this as I work on that, since anything you can do in `quickfort` you should be able to do manually with a UI like this issue describes.
